functions/APIs/surplus.js

```diff
@@ -0,0 +1,175 @@
+const { db } = require('../util/admin');
+
+/*
+display a list of all surplus
+GET /surplus
+success response: array of surplus objects
+*/
+exports.getAllSurplus = (request, response) => {
+    db.collection('surplus').get()
+    .then((data) => {
+        let surplus = [];
+        data.forEach((doc) => {
+            surplus.push(doc);
+        });
+
+        //get produce object that is linked to surplus object
+        function queryProduce(produceId) {
+            return new Promise(function(resolve, reject) {
+                db.doc(`/produce/${produceId}`).get()
+                .then((doc) => {
+                    resolve(doc.data());
+                })
+                .catch((err) => {
+                    return reject(err);
+                });
+            });
+        }
+
+        //get farm object that is linked to surplus object
+        function queryFarms(originFarmId) {
+            return new Promise(function(resolve, reject) {
+                db.doc(`/farms/${originFarmId}`).get()
+                .then((doc) => {
+                    resolve(doc.data());
+                })
+                .catch((err) => {
+                    return reject(err);
+                });
+            });
+        }
+
+        //get produce and farm objects that are linked to surplus object
```

Comment:
Perhaps some bigger picture context on the usage of this query could be helpful, to make it clear that we will display this info for surplus objects and use it to create the deal object/Freight API calls